lib/ansible/modules/database/postgresql/postgresql_query.py

```diff
@@ -0,0 +1,347 @@
+     stored in the DB encrypted will be assumed to have changed.
+   - If you specify PUBLIC as the user, then the privilege changes will apply
+     to all users. You may not specify password or role_attr_flags when the
+     PUBLIC user is specified.
+   - The ssl_rootcert parameter requires at least Postgres version 8.4 and I(psycopg2) version 2.4.3.
+requirements: [ psycopg2 ]
+author:
+    - "Felix Archambault (@archf)"
+    - "Will Rouesnel (@wrouesnel)"
+'''
+
+EXAMPLES = '''
+# Insert or update a record in a table with positional arguments
+- postgresql_query:
+    db: acme
+    login_user: django
+    login_password: ceec4eif7ya
+    query: SELECT * FROM a_table WHERE a_column=%s AND b_column=%s
+    positional_args:
+      - "positional string value 1"
+      - "positional string value 2"
+
+# Insert or update a record in a table with named arguments
+- postgresql_query:
+    db: acme
+    login_user: django
+    login_password: ceec4eif7ya
+    query: SELECT * FROM some_table WHERE a_column=%(a_value)s AND b_column=%(b_value)s
+    positional_args:
+      a_value: "positional string value 1"
+      b_value: "positional string value 2"
+
+
+# Run queries from a '.sql' file
+- postgresql_query:
+    db: acme
+    login_user: django
+    login_password: ceec4eif7ya
+    query: "{{playbook_dir}}/scripts/my_sql_query_file.sql"
+    named_args:
+      a_value: "positional string value 1"
+      b_value: "positional string value 2"
+
+# Run queries from a '.sql' file and assign result in a fact available for the
+# rest of the ansible runtime. Query inside scripts may contain 'named_args'
+# or 'positional_args'.
+- postgresql_query:
+    db: acme
+    login_user: django
+    login_password: ceec4eif7ya
+    query: SELECT * FROM some_table WHERE a_column=%(a_value)s AND b_column=%(b_value)s
+    query: /path/to/my/script.sql
+    named_args:
+      a_value: "positional string value 1"
+      b_value: "positional string value 2"
+    fact: my_key
+'''
+
+RETURN = '''
+query_results:
+    description: List of dictionaries in column:value form representing returned rows.
+    returned: changed
+    type: list
+    sample: [{"Column": "Value1"},{"Column": "Value2"}]
+row_count:
+    description: number of affected rows by query, if applicable
+    returned: changed
+    type: int
+    sample: 5
+'''
+
+HAS_PSYCOPG2 = False
+try:
+    import psycopg2
+    import psycopg2.extras
+except ImportError:
+    pass
+else:
+    HAS_PSYCOPG2 = True
+
+import traceback
+from ansible.module_utils.six import iteritems
+from ansible.module_utils.basic import AnsibleModule, get_exception, BOOLEANS
+import ansible.module_utils.postgres as pgutils
+
+
+def main():
+    argument_spec = pgutils.postgres_common_argument_spec()
+
+    argument_spec.update(dict(
+        db=dict(default=None),
+        query=dict(type="str"),
+        autocommit=dict(type='bool', default=False, choices=BOOLEANS),
```

Comment:
I needed to patch this line on the latest devel head to this in order to let the module work.
```
        autocommit=dict(type='bool', default=False, choices=tuple(BOOLEANS)),
```